test/e2e/network/networking.go

```diff
@@ -321,39 +333,83 @@ var _ = SIGDescribe("Networking", func() {
 			ginkgo.By(fmt.Sprintf("dialing(udp) %v --> %v:%v (config.clusterIP)", config.TestContainerPod.Name, config.ClusterIP, e2enetwork.ClusterUDPPort))
 			err := config.DialFromTestContainer("udp", config.ClusterIP, e2enetwork.ClusterUDPPort, config.MaxTries, 0, config.EndpointHostnames())
 			if err != nil {
-				framework.Failf("failed dialing endpoint, %v", err)
+				framework.Failf("failed dialing endpoint (initial), %v", err)
 			}
+			ginkgo.By("Deleting a pod which, will be replaced with a new endpoint")
 			config.DeleteNetProxyPod()
 
-			ginkgo.By(fmt.Sprintf("dialing(udp) %v --> %v:%v (config.clusterIP)", config.TestContainerPod.Name, config.ClusterIP, e2enetwork.ClusterUDPPort))
+			ginkgo.By(fmt.Sprintf("dialing(udp) %v --> %v:%v (config.clusterIP) (endpoint recovery)", config.TestContainerPod.Name, config.ClusterIP, e2enetwork.ClusterUDPPort))
 			err = config.DialFromTestContainer("udp", config.ClusterIP, e2enetwork.ClusterUDPPort, config.MaxTries, config.MaxTries, config.EndpointHostnames())
 			if err != nil {
-				framework.Failf("failed dialing endpoint, %v", err)
+				framework.Failf("failed dialing endpoint (recovery), %v", err)
+			}
+		})
+
+		// canCreateNodePortService is a helper function to validate nodeport functionality is working at a baseline, supports http/udp,
+		// this was added so that we can separate testing of XYZ updates from just plain old XYZ nodeport service functioning.
+		canCreateNodePortService := func(proto string, config *e2enetwork.NetworkingTestConfig) error {
```

Comment:
for e2e too much abstraction is a pain for debugging test failures, I'd rather prefer to copy and paste, in terms of LOC it seems the same number

Reply:
Counterpoint - positive controls should be declarative, unique functionality (the experiment that the test is measuring) should be imperative.

But if you feel strongly I can back this out and copy the code .  Still making coffee so Brain isn't fully powered yet :)

Reply:
tell me that once you spend 1 week trying to deflake an e2e test and found that someone refactored the function without knowing that affected the test ;)
e2e should be simple enough that anybody with a minimum level is able to look at it and fix/add new ones, this is not adding much value compared to add the lines in each test.

Reply:
ok i assume that you feel strongly then on this one :)

Reply:
ok updated to c/p version